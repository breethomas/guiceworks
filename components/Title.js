// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import SerifBold from './SerifBold'

type Props = {
  tag?: string,
}

const style = css(
  { fontSize: 40 },
  media2({ fontSize: 80 }),
)

const Title = ({ tag, ...props }: Props) => (
  <SerifBold className={style} tag={tag} {...props} />
)

Title.defaultProps = {
  tag: 'h1',
}

export default Title
