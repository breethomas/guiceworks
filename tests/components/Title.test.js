import React from 'react'
import renderer from 'react-test-renderer'
import Title from '../../components/Title'

describe('Title', () => {
  it('shows the correct snapshot for the Title component tree (default)', () => {
    const component = <Title>Title</Title>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Title component tree (tag)', () => {
    const component = <Title tag="span">Title</Title>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
