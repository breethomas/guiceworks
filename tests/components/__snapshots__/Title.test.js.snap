// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title shows the correct snapshot for the Title component tree (default) 1`] = `
<h1
  className="css-lcomo css-5eu6oy"
>
  Title
</h1>
`;

exports[`Title shows the correct snapshot for the Title component tree (tag) 1`] = `
<span
  className="css-lcomo css-5eu6oy"
>
  Title
</span>
`;
